test/common/testlib.py

```diff
@@ -784,9 +785,22 @@ def enable_preload(self, package, *pages):
         self.write_file(path, '{ "preload": [%s]}' % ', '.join('"{0}"'.format(page) for page in pages))
 
     def setUp(self, restrict=True):
+        if os.getenv("MACHINE"):
+            # apply env variable together if MACHINE envvar is set
+            opts.address = os.getenv("MACHINE")
+            if os.getenv("DESTRUCTIVE") and not self.is_nondestructive():
```

Comment:
This is wrong -- you'll skip  *every* test if `$DESTRUCTIVE` is not set.

Reply:
nope, meaning of that should be to run also destructive tests if you want. but yep, maybe I've done mistake there, although I tried to tests these corner cases,  that's true that maybe I've test something before I've changed this line, so I did not catch the issue the now it really does not do what I wantet :-)

Reply:
I meant that if `DESTRUCTIVE` is not set, it's always going to the `else:` branch and thus do SkipTest.